### PR TITLE
jenkins: fix workspace cleanup

### PR DESCRIPTION
**Pull request by jnewb1**
use a class for common Paths, similar to the C++ "Path" namespace (but we have to call it "Paths" to prevent conflict with pathlib.Path

### CODE REVIEW

File: common/prefix.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 class OpenpilotPrefix:
   def __init__(self, prefix: Optional[str] = None, clean_dirs_on_exit: bool = True):
-    self.prefix = prefix if prefix else str(uuid.uuid4())
+    self.prefix = prefix if prefix else str(uuid.uuid4().hex[0:15])
```
Comment (jnewb1) [inline common/prefix.py:11]: the C++ implementation uses 15 alphanumeric characters, and I think that looks better than the uuid format with dashes
'550e8400-e29b-41d4-a716-446655440000'

File: common/util.h [hunk 2]
```diff
@@ -188,9 +188,9 @@ class LogState {
   void *zctx = nullptr;
   void *sock = nullptr;
   int print_level;
-  const char* endpoint;
+  std::string endpoint;
 
-  LogState(const char* _endpoint) {
+  LogState(std::string _endpoint) {
```
Comment (jnewb1) [inline common/util.h:193]: using a c string here causes issues because swaglog_ipc returns a dynamically generated string

File: conftest.py [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
 from openpilot.common.prefix import OpenpilotPrefix
+from openpilot.system.hardware.hw import Paths
 
 
 @pytest.fixture(scope="function", autouse=True)
 def global_setup_and_teardown():
   # setup a clean environment for each test
   with OpenpilotPrefix():
+    os.makedirs(Paths.log_root())
```
Comment (adeebshihadeh) [inline conftest.py:13]: this should be in the prefix

**Review by adeebshihadeh — COMMENTED** [review]
I think it's an acceptable tradeoff for this PR, but the Paths stuff really shouldn't go in HW or even system/. We should start thinking about cleaning that up

File: system/hardware/hw.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import os
+from selfdrive.hardware import PC
+from pathlib import Path
```
Comment (adeebshihadeh) [inline system/hardware/hw.py:3]: imports should go stdlib + third party, then our imports

File: conftest.py (hunk 3 above)
Comment (jnewb1) [reply conftest.py:13]: it is in the prefix (since comma_home has the prefix), but the folder isn't created automatically

Comment (jnewb1) [reply conftest.py:13]: so its like /home/batman/.comma12345678/log now

Comment (adeebshihadeh) [reply conftest.py:13]: it should be handled in the prefix...

File: system/hardware/hw.py (hunk 4 above)
Comment (jnewb1) [reply system/hardware/hw.py]: done

File: conftest.py (hunk 3 above)
Comment (jnewb1) [reply conftest.py]: done

**Comment by jnewb1** [comment]
leaving open so we can identify the jenkins issue

**Pull request closed without merging**